hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -122,68 +446,81 @@ mockChainSyncServer queue =
         body = toTxSeq $ StrictSeq.singleton tx
      in BlockAlonzo $ mkShelleyBlock $ Ledger.Block header body
 
-  serverStIdle :: ServerStIdle Block (Point Block) (Tip Block) m ()
-  serverStIdle =
+  serverStIdle :: Int -> ServerStIdle Block (Point Block) (Tip Block) m ()
+  serverStIdle cursor =
```

Comment:
Not sure if necessary here as this might be forced otherwise, but `!cursor` is something I always do when there is a recursion and `+ 1` :)